code/obj/artifacts/artifact_items/handheld_storage.dm

```diff
@@ -0,0 +1,180 @@
+		src.wear_layer = initial(src.wear_layer)
+
+		// reapply overlay if needed
+		src.UpdateOverlays(src.wizard_gem_image, "gem")
+		src.wizard_gem_image = null
+
+	// used for martian bag of holding changing size
+	proc/martian_change_shape()
+		var/list/prev_contents = list()
+		for (var/atom/A as anything in src.storage?.get_contents())
+			prev_contents += A
+			src.storage.transfer_stored_item(A, get_turf(src))
+
+		src.create_storage(/datum/storage/artifact_bag_of_holding/martian, max_wclass = pick(W_CLASS_TINY, W_CLASS_SMALL, W_CLASS_NORMAL, W_CLASS_BULKY),
+			slots = rand(3, 13), opens_if_worn = TRUE)
+
+		for (var/atom/A as anything in prev_contents)
+			if (src.storage.check_can_hold(A) == STORAGE_CAN_HOLD)
+				src.storage.add_contents(A, null, FALSE)
+
+/datum/artifact/bag_of_holding
+	associated_object = /obj/item/artifact/bag_of_holding
+	type_name = "Bag of Holding"
+	rarity_weight = 200
+	validtypes = list("eldritch", "martian", "wizard")
+	validtriggers = list(/datum/artifact_trigger/force, /datum/artifact_trigger/electric, /datum/artifact_trigger/heat,
+		/datum/artifact_trigger/radiation, /datum/artifact_trigger/carbon_touch, /datum/artifact_trigger/silicon_touch,
+		/datum/artifact_trigger/cold)
+	type_size = ARTIFACT_SIZE_MEDIUM
+	react_xray = list(5, 91, 97, 11, "HOLLOW")
+
+	effect_activate(obj/O)
+		if (..())
+			return TRUE
+
+		var/obj/item/artifact/bag_of_holding/boh = O
+
+		switch(src.artitype.name)
+			// large variety of storage, items are stored in LIFO, FIFO, or random order
+			if ("eldritch")
+				var/slots = rand(3, 13)
+				var/wclass
+
+				// pick max w class that can be held
+				if (slots >= 3 && slots <= 5)
+					wclass = pick(prob(10); W_CLASS_TINY, prob(50); W_CLASS_SMALL, prob(100); W_CLASS_NORMAL, prob(40); W_CLASS_BULKY)
+				else if (slots > 5 && slots <= 9)
+					wclass = pick(prob(20); W_CLASS_TINY, prob(100); W_CLASS_SMALL, prob(50); W_CLASS_NORMAL)
+				else if (slots >= 9)
+					wclass = pick(prob(30); W_CLASS_TINY, prob(100); W_CLASS_SMALL, prob(10); W_CLASS_NORMAL)
+
+				// allow it to be worn if it can hold a lot of items
+				//if (slots > 4)
+				//	boh.c_flags |= ONBELT
+				if (slots > 6)
+					boh.c_flags |= ONBACK
+
+				// set weight size if its a large storage
+				if ((slots > 8 || wclass == W_CLASS_BULKY) && prob(90))
+					boh.w_class = W_CLASS_BULKY
+
+				boh.create_storage(/datum/storage/no_hud/eldritch_bag_of_holding, max_wclass = wclass, slots = slots, opens_if_worn = \
+					boh.c_flags & ONBELT || boh.c_flags & ONBACK, params = list("use_inventory_counter" = TRUE, "item_pick_type" = \
+					pick(STORAGE_NO_HUD_QUEUE, STORAGE_NO_HUD_STACK, STORAGE_NO_HUD_RANDOM)))
+
+			// immediately start the storage effect, can always be worn
+			if ("martian")
+				boh.c_flags |= (ONBELT | ONBACK)
+				boh.martian_change_shape()
+
+			// "large sized" storage that can hold small items
+			if ("wizard")
+				boh.w_class = W_CLASS_BULKY
+				//boh.c_flags |= (ONBELT | ONBACK)
```

Comment:
?

Reply:
Place holder for if belt gets added, I'll just remove for now